Felienne/hedy
# [BUG FIX] Login re-direct on class pre-join works as expected

**Description**
In this PR we fix an issue where the login re-direct on the class pre-join was broken. That was due to two separate issues:

- We broken `auth.profile` with #2101
- The saved re-direct item called a `link` value that didn't exist, it should have been `id`

Both are fix in this PR and the temporary storing of the join link is fixed again.

**Fixes**
No related issue

**How to test**
Create a class (or use an already existing one) and make sure you have an account that is not already in this class. Copy the join link, log out and navigate to the join link. Notice the pop-up "you can only join if logged in, do you want to join?". Log in and you should automatically join the class (as the link is temporary stored). You can verify this my navigating to 'My account' -> 'My classes'.

**Checklist**
Done? Check if you have it all in place using this list*
  

* If you're unsure about any of these, don't hesitate to ask. We're here to help!
